react/getting_started_with_react/keys_in_react.md

```diff
@@ -0,0 +1,126 @@
+### Introduction
+
+In this lesson we will cover keys in React. Keys are special props for our components and we'll learn why they are used.
+
+### Lesson overview
+
+By the end of the lesson you should be able to answer the following:
+
+- What are keys and why does React need them?
+- What are good and bad examples of keys?
+
+### Why does React need keys?
+
+In the upcoming lessons as you learn more about the internal workings of React, more specifically the rerendering process, you will understand the importance of keys. For now we will keep it simple. 
+
+In the previous lesson on rendering lists we used the `.map()` method to iterate over an array of data and return a list of elements. Now imagine, if any of the item in the list were to change, how would React know which item to update?
+
+If the list were to change, one of two things _should_ happen:
+1. we completely re-render the entire list, or:
+2. we hunt down that specific items that were changed and only re-render those.
+
+Assuming we want to hunt down that one specific item that was changed and NOT re-render the entire list. We need something to track that specific item. We can track down a specific item by using a `key`.
+
+When the list is updated for whatever reason, (either from a server or a user interaction), React matches the `keys` of each of the previous list to the updated list. If there were any changes, React will only update the items that have changed.
+
+As long as `keys` remain consistent and unique, React can handle the DOM effectively and efficiently.
+
+### Using keys
+
+Keys are passed into the component or a DOM element as a prop. You should already be familiar with the syntax.
+
+~~~jsx  
+<Component key={keyValue} />
+//or
+<div key={keyValue} />
+~~~
+
+<span id="keys-from-data">Now that we know the syntax, the next question is: what should be used as a key? Ideally they should be some identifier that is unique to each item in the list. Most databases assign a unique id to each entry, so you shouldn't have to worry about assigning and id yourself. If you are defining data yourself, it is good practice to assign a unique `id` to each item. You may use the [uuid](https://www.npmjs.com/package/uuid) package to generate a unique id. Let's look at an example:</span>
```

Comment:
Personally I feel like we could have used a non-3rd party package to show how to pass in keys, at least at first. For the purposes of the example it probably makes sense, but we could have as easily hardcoded unique `id` values here, then gone into using something like UUID and not generating keys on the fly further down.